src/printer.js

```diff
@@ -621,7 +621,40 @@ function genericPrintNoParens(path, options, print) {
         } else {
           parts.push(printPropertyKey(path, print));
         }
-        parts.push(": ", path.call(print, "value"));
+        let noBreak = [
+          "ArrayExpression",
+          "ArrayPattern",
+          "ArrowFunctionExpression",
+          "AssignmentPattern",
+          "CallExpression",
+          "FunctionExpression",
+          "Identifier",
+          "Literal",
+          "LogicalExpression",
```

Comment:
(Since the original issue was that binary expressions would break weird, we probably want to allow the expression to be moved down for both binary and logical, so you should probably remove LogicalExpression here)